include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -119,7 +123,18 @@ class sycl_backend
                         s.report(execution_info::task_completion);
                     }
                     if constexpr (report_value_v<SelectionHandle, execution_info::task_time_t>)
-                        s.report(execution_info::task_time, (std::chrono::steady_clock::now() - t0).count());
+                    {
+                        if (use_event_profiling)
+                        {
+                            cl_ulong time_start = e1.template get_profiling_info<sycl::info::event_profiling::command_start>();
```

Comment:
This looks fine given current issues with command_submit.

Reply:
Discussed offline: Current approach is relying on undefined behavior.
Spec doesn't allow capturing or using SYCL class other then accessors in a [`host_task`](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:interfaces.hosttasks)
How about storing events, e.g. in a `std::vector`, and process them later?